kncaputo/underrated
# Cleanup/dropdown button and styles

## What's this PR do?
- Creates `Header` component - closes #32 
- Adds helper function in `App` to create dropdown buttons for genres and ratings filters - closes #33
- Add `Watchlist` component - closes #34
- Switch css to scss - closes #30
- Fix issue with conditional rendering for `clear` button that shows when using filters - closes #28 
- Adds styles to dropdowns - closes #29 
- Adds margin to movie details page for small screens
- Removes movie banner on home page for small screens

## Where should the reviewer start?
- `Header` 
- `generateGenreButtons` and `generateRatingsButtons` in `App`
- Review scss styles

## How should this be manually tested?
- Test in browser by trying out all filter options. Clear button should render as soon as any filter is implemented. Try logging in and out. Check styles. Check on mobile devices.

## Any background context you want to provide?
- Still unable to remove the blue from around the dropdown buttons.
- Background music plays on mobile view even though the trailer banner is deleted... 

## Screenshots/GIFs
<img width="1015" alt="Screen Shot 2021-01-25 at 6 47 55 PM" src="https://user-images.githubusercontent.com/68157759/105780243-074c8b00-5f3e-11eb-9b24-0a294bc7507d.png">

<img width="408" alt="Screen Shot 2021-01-25 at 6 48 03 PM" src="https://user-images.githubusercontent.com/68157759/105780245-087db800-5f3e-11eb-9a6a-8ef3111d30f8.png">

<img width="252" alt="Screen Shot 2021-01-25 at 10 41 46 PM" src="https://user-images.githubusercontent.com/68157759/105797792-ac775b80-5f5e-11eb-8032-6934fab8aae7.png">


## Questions
- @lucyconklin We are using bootstrap dropdown menus in three different places on the page - in the `Login` component, for search by genre, and search by rating. Is there a way to target them differently for styles? So far we have only had success targeting the dropdown using the class `.dropdown-menu.show`. It seems to be a built in class. When we've tried to assign a class, it seems to be a layer above the actual element. The problem we are currently running into targeting them the way we are, is that all of our dropdowns are now a fixed height because we need one to be a fixed height.
- A similar issue, we are having trouble removing the blue box that appears when the button is clicked to drop the dropdown menu. I think figuring out the previous question would also apply to this. There's a photo example in the screenshot of the dropdown menu above.
